app/backend/approaches/readretrieveread.py

```diff
@@ -108,5 +108,5 @@ def __init__(self, employee_name: str):
         self.func = self.employee_info
         self.employee_name = employee_name
 
-    def employee_info(self, unused: str) -> str:
-        return self.lookup(self.employee_name)
+    def employee_info(self, name: str) -> str:
+        return self.lookup(name)
```

Comment:
This makes sense, please change the type signature of the base method (`lookup`) to be `str` instead of `Optional[str]` since it will always return a str and if you're just returning the value of `lookup` then the signature should match that of `func` which is `Callable[..., str]`

https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/app/backend/lookuptool.py#L19

Reply:
I'll address this post #339